benchmarks/transformer/sdpa.py

```diff
@@ -61,11 +74,52 @@ class Experiment:
     results: ExperimentResults
 
     def asdict(self):
-        dict1 = asdict(self.config)
-        dict2 = asdict(self.results)
+        dict1 = self.config.asdict()
+        dict2 = self.results.asdict()
         return {**dict1, **dict2}
 
 
+def calculate_tflops(
+    config: ExperimentConfig, time_us: float, is_backward: bool = False
+) -> float:
+    """
+    Calculate TFLOPS for scaled dot product attention.
+
+    Parameters:
+    - config: The experiment configuration
+    - time_us: The execution time in microseconds
+    - is_backward: Whether to calculate for backward pass (includes gradient computation)
+
+    Returns:
+    - TFLOPS value
+    """
+    B = config.batch_size
+    H = config.num_heads
+    M = config.q_seq_len
+    N = config.kv_seq_len
+    D = config.head_dim
+
+    # Forward pass FLOPs
+    qk_flops = (
+        M * N * D * 2
+    )  # Q*K^T matmul: M×D times D×N = M×N with 2 FLOPs per multiply-add
```

Comment:
super nit / optional: i find this notation for expressing a matmul confusing, imo `(M,D) @ (D,N) = (M,N)` or similar is clearer.